core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -3159,4 +3160,44 @@ private SqlSpecialOperatorWithPolicy(String name, SqlKind kind, int prec, boolea
   @Test void testSimplifyVarbinary() {
     checkSimplifyUnchanged(cast(cast(vInt(), tVarchar(true, 100)), tVarbinary(true)));
   }
+
+  @Test public void testEliminateCommonExprInCondition() {
+    RexSimplify rexSimplify = new RexSimplify(rexBuilder, RelOptPredicateList.EMPTY, EXECUTOR);
+
+    // (a=b && c=7) || (a=b && c=7 && d=8) => (a=b && c=7)
+    RexNode or1 = or(
+        and(vBool(0), vBool(1)),
+        and(vBool(0), vBool(1), vBool(2))
+    );
+    RexNode orSimplified2 = rexSimplify.eliminateCommonExprInCondition(or1);
```

Comment:
minor: I guess this should be named `orSimplified1` (and the one below in line 3182 should be `orSimplified2`)

Reply:
😂 Revised.